src/CliCommand/SoxCommand.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace IndieHD\AudioManipulator\CliCommand;
+
+use IndieHD\AudioManipulator\CliCommand\SoxCommandInterface;
+
+class SoxCommand extends CliCommand implements SoxCommandInterface
+{
+    private $name = 'sox_command';
+
+    protected $binary = 'sox';
+
+    protected $parts = [
```

Comment:
Make this an over-writable property in the base class

Reply:
What should its initial value be? An empty array, e.g., `protected $parts = [];`?

Reply:
make it an abstract property (forcing the implementation to overwrite it) and yes an empty array

Reply:
ignore that 😕

Reply:
yes initialize it to an empty array or `null` ? if `null` then we can `test` if its been been set in the implementation, if not through an exception